dspace-api/src/main/java/org/dspace/storage/bitstore/AzureBlobStorageBitStoreService.java

```diff
@@ -0,0 +1,395 @@
+	 * @return attrs A Map with key/value pairs of desired metadata If file not
+	 *         found, then return null
+	 */
+	@Override
+	public Map about(Bitstream bitstream, Map attrs) throws IOException {
+		String key = getFullKey(bitstream.getInternalId());
+		try {
+			CloudBlob blob = this.blobContainer.getBlockBlobReference(key);
+			blob.downloadAttributes();
+
+			if (blob != null) {
+				BlobProperties blobProps = blob.getProperties();
+
+				if (blobProps != null) {
+					Date lastModified = blobProps.getLastModified();
+					String lenght = String.valueOf(blobProps.getLength());
+					String checksum = blobProps.getEtag();
+					String lastModifiedTimeString = null;
+
+					if (lastModified != null) {
+						lastModifiedTimeString = String.valueOf(lastModified.getTime());
+					}
+
+					String newLine = System.getProperty("file.separator");
+
+					log.debug("blobProperties:" + newLine + "lenght: " + lenght + newLine + "checksum: " + checksum
+							+ newLine + "last modified: " + lastModifiedTimeString);
+
+					log.debug("Attributes raw: " + attrs);
+
+					if (attrs != null) {
+
+						log.debug("Attributes as String: " + attrs.toString());
+
+						if (lenght != null && attrs.containsKey("size_bytes")) {
+							attrs.put("size_bytes", lenght);
+						}
+
+						if (checksum != null && attrs.containsKey("checksum")) {
+							attrs.put("checksum", checksum);
+							attrs.put("checksum_algorithm", this.algorithm);
+						}
+
+						if (lastModifiedTimeString != null && attrs.containsKey("modified")) {
+							attrs.put("modified", lastModifiedTimeString);
+						}
+
+						return attrs;
+					} else {
+						// not sure if this would ever happen
+						return null;
+					}
+				}
+			}
+
+		} catch (Exception e) {
+			log.error("about(" + key + ", attrs)", e);
+			throw new IOException(e);
+		}
+
+		return null;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.dspace.storage.bitstore.BitStoreService#remove(org.dspace.content.
+	 * Bitstream)
+	 */
+	@Override
+	public void remove(Bitstream bitstream) throws IOException {
+
+		String key = getFullKey(bitstream.getInternalId());
+		log.info("Attempting removal of Azure Blob storage file: " + key);
+
+		try {
+			CloudBlob blob = this.blobContainer.getBlockBlobReference(key);
+			if (blob.exists()) {
+				blob.delete();
+				if (!blob.exists()) {
+					log.info("Removal of Azure Blob storage file: " + key + " was successful");
+				}
+			} else {
+				log.warn("File didn't exist in Azure Blob Storage " + key);
+			}
+		} catch (Exception e) {
+			log.error("remove(" + key + ")", e);
+			throw new IOException(e);
+		}
+
+	}
+
+	/**
+	 * Creates and returns a temporary local file for use by the sample.
+	 *
+	 * @param blockBlob
+	 *            CloudBlockBlob object.
+	 * @param filePath
+	 *            The path to the file to be uploaded.
+	 *
+	 * @throws Throwable
+	 */
+	private void uploadFileBlocksAsBlockBlob(CloudBlockBlob blockBlob, String filePath) throws Throwable {
```

Comment:
`throws Throwable` is not recommended. I'd rather see several `throws` here with the individual exception types listed out (and then the Javadocs can note when specific Exception types may occur)